extensions/api/data-management/transferprocess/src/main/java/org/eclipse/dataspaceconnector/api/datamanagement/transferprocess/service/TransferProcessServiceImpl.java

```diff
@@ -0,0 +1,114 @@
+
+    public TransferProcessServiceImpl(TransferProcessStore transferProcessStore, TransferProcessManager manager, TransactionContext transactionContext) {
+        this.transferProcessStore = transferProcessStore;
+        this.manager = manager;
+        this.transactionContext = transactionContext;
+    }
+
+    @Override
+    public @Nullable TransferProcess findById(String transferProcessId) {
+        return transactionContext.execute(() -> transferProcessStore.find(transferProcessId));
+    }
+
+    @Override
+    public @NotNull Collection<TransferProcess> query(QuerySpec query) {
+        return transactionContext.execute(() -> transferProcessStore.findAll(query).collect(toList()));
+    }
+
+    @Override
+    public @Nullable String getState(String transferProcessId) {
+        return transactionContext.execute(() -> {
+            var process = transferProcessStore.find(transferProcessId);
+            if (process == null) {
+                return null;
+            }
+            return getStateName(process);
+        });
+    }
+
+    @Override
+    public @NotNull Result<?> cancel(String transferProcessId) {
+        return transactionContext.execute(() -> {
+            var transferProcess = transferProcessStore.find(transferProcessId);
+
+            if (transferProcess == null) {
+                return Result.failure("Not found " + transferProcessId);
+            }
+
+            // Attempt the transition only to verify that the transition is allowed.
+            // The updated transfer process is not persisted at this point, and is discarded.
+            try {
+                transferProcess.transitionCancelled();
+            } catch (IllegalStateException e) {
+                return Result.failure("Cannot cancel a transfer process in state " + getStateName(transferProcess));
+            }
+
+            manager.enqueueCommand(new CancelTransferCommand(transferProcessId));
+
+            return Result.success();
+        });
+    }
+
+    @Override
+    public @NotNull Result<?> deprovision(String transferProcessId) {
+        return transactionContext.execute(() -> {
+            var transferProcess = transferProcessStore.find(transferProcessId);
+
+            if (transferProcess == null) {
+                return Result.failure("Not found " + transferProcessId);
+            }
+
+            // Attempt the transition only to verify that the transition is allowed.
+            // The updated transfer process is not persisted at this point, and is discarded.
+            try {
+                transferProcess.transitionDeprovisioning();
+            } catch (IllegalStateException e) {
+                return Result.failure("Cannot deprovision a transfer process in state " + getStateName(transferProcess));
+            }
+
+            manager.enqueueCommand(new DeprovisionRequest(transferProcessId));
+
+            return Result.success();
+        });
+    }
+
+    @NotNull
```

Comment:
For consistency place the @NotNull annotation either before the method, or before the return value in all methods.